code/modules/projectiles/guns/energy.dm

```diff
@@ -593,4 +546,359 @@
 	var/obj/screen/ammo/A = user.hud_used.ammo //The ammo HUD
 	A.update_hud(user)
 
+/obj/item/weapon/gun/energy/lasgun/lasrifle/update_item_state(mob/user) //Without this icon states for wielded guns won't show.
+	. = item_state
+	item_state = "[initial(icon_state)][flags_item & WIELDED ? "_w" : ""]"
+
```

Comment:
This does not update the inhand icons as intended and instead just does the same thing it does by default

Reply:
As stated in discord, removing or editing this any further breaks it and wielded icons will not show

Reply:
see review comments